src/main/java/io/aiven/kafka/connect/http/sender/AbstractHttpSender.java

```diff
@@ -47,6 +47,26 @@ protected AbstractHttpSender(
         this.httpClient = Objects.requireNonNull(httpClient);
     }
 
+    public final HttpResponse<String> send(final String body, final String key) {
+        if (this.config.httpUri().toString().contains("{key}")) {
+            final var configOriginalMap = this.config.originalsStrings();
+            final String genericUrl = configOriginalMap.get("http.url");
+            final String newUrl = genericUrl.replace("{key}", key);
+            configOriginalMap.replace("http.url", newUrl);
+            log.debug("Key replaced in URL. New URL: {}", newUrl);
+
+            final var newConfig = new HttpSinkConfig(configOriginalMap);
+            log.debug("Sending request with body: {}", body);
```

Comment:
Instead of modifying the configs on every execution, I'd prefer a more structured approach to flag this.
Consider:
- adding a new config, e.g. `http.url.inject.key.enable=true/false` defaults: false.
- maybe have a fallback value to use it when key is null or cannot be converted, e.g. `http.url.inject.key.fallback=string` defaults: null -- if value cannot be converted or null, then fail at runtime. 
- delegate to the config on how to replace the key with the url by adding a new `config.httpUri(keySupplier.get())`